tasks/main.yml

```diff
@@ -1,6 +1,10 @@
 ---
 # File: main.yml - Main tasks for Nomad
 
+- name: Merge default and custom config
+  set_fact:
+    nomad_config_merged: "{{ nomad_config_defaults | combine((nomad_config | default({})), recursive=True) }}"
```

Comment:
Two things, you could do this without `set_fact` by putting this into `defaults/main.yml`.

You could handle this in a backwards compatible way by checking the type of the `nomad_config ` variable, and if it's a string, you handle it using the old logic, otherwise if it's an object, use the new logic. Possibly just use a few temp variables in `defaults/main.yml` (this is untested):

```
__nomad_config: "{{ nomad_config if is nomad_config mapping else (nomad_config | from_json) }}"
nomad_config_merged: "{{ nomad_config_defaults | combine((__nomad_config | default({})), recursive=True) }}"
```